### PR TITLE
Handle schema application more gracefully when provided a bad list

### LINKED ISSUE #212: Schema phase raises exception when [Int] given for expected Int input
The schema phase gives the following exception:

```
     ** (KeyError) key :of_type not found in: %Absinthe.Type.Scalar{__private__: [], __reference__: %{identifier: :integer, location: %{file: "/Users/bruce/Code/CS/absinthe/lib/absinthe/type/built_ins/scalars.ex", line: 6}, module: Absinthe.Type.BuiltIns.Scalars}, description: "The `Int` scalar type represents non-fractional signed whole numeric values.\nInt can represent values between `-(2^53 - 1)` and `2^53 - 1` since it is\nrepresented in JSON as double-precision floating point numbers specified\nby [IEEE 754](http://en.wikipedia.org/wiki/IEEE_floating_point).", name: "Int", parse: #Function<10.65559679/1 in Absinthe.Type.BuiltIns.Scalars.parse_with/2>, serialize: #Function<9.65559679/1 in Absinthe.Type.BuiltIns.Scalars.__absinthe_type__/1>}
```

When attempting to run this test:

```elixir
defmodule Absinthe.Phase.SchemaTest do
  use Absinthe.Case, async: true

  defmodule IntegerInputSchema do
    use Absinthe.Schema

    query do

      field :test, :string do
        arg :integer, :integer
        resolve fn
          _, args, _ ->
            {:ok, "ayup"}
        end
      end

    end

  end


  @query """
  { test(integer: [1]) }
  """

  it "gives a useful error message" do
    assert {:ok, _} = run(@query)
  end

  def run(query) do
    with {:ok, value} <- Absinthe.Phase.Parse.run(query),
         {:ok, value} <- Absinthe.Phase.Blueprint.run(value) do
      Absinthe.Phase.Schema.run(value, schema: IntegerInputSchema)
    end
  end

end
```

My best guess is that this is a recent regression.

### PR DESCRIPTION
**Pull request by bruce**
Closes #212

### CODE REVIEW

**Comment by benwilson512** [comment]
I don't think this is right. It shouldn't be getting to the resolution phase at all. Giving a list instead of a scalar should be a validation error, and it should jump from the validation result phase.

**Comment by bruce** [comment]
This is just schema application. This is pre-validation.

File: test/lib/absinthe/phase/schema_test.exs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    end
+
+
+    @query """
+    { test(integer: [1]) }
+    """
+
+    it "doesn't raise an exception" do
+      assert {:ok, _} = run(@query)
```
Comment (benwilson512) [inline test/lib/absinthe/phase/schema_test.exs]: We should be testing for an error here right?

Comment (bruce) [inline test/lib/absinthe/phase/schema_test.exs]: No, this is the result of the phase -- see the definition of `run` below.

**Comment by benwilson512** [comment]
I'm dumb, ignore me....

**Pull request merged**